provider/resource_harbor_robot_account.go

```diff
@@ -177,7 +177,7 @@ func resourceRobotAccountCreate(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
-	return resourceRobotAccountRead(d, meta)
+	return resourceRobotAccountUpdate(d, meta)
```

Comment:
what was the reason for this change?  it seems odd to me to create then immediately update here.

Reply:
It's caused by an issue with the Harbor API design. The POST API call doesn't accept 'disabled' as a parameter, but the PUT API call does. Naturally, the Terraform resource doesn't distinguish, which can cause an issue on initial creation where a second Terraform run was needed if it's desired to create a robot account in a disabled state. This change fixes that.

An alternative solution would be to call the update from the client in this code block, as opposed to calling out to resourceUpdate. I think that solution may be slightly more readable, but duplicates some code so I slightly preferred this solution. Very open to changing it though.